solana/solana-ibc/programs/solana-ibc/src/storage.rs

```diff
@@ -352,12 +352,34 @@ pub fn get_provable_from<'a, 'info>(
     get(info).map_err(|err| err.with_account_name("trie"))
 }
 
+/// Used for finding the account info from the keys.
+/// Useful for finding the token mint on the source chain which cannot be derived from the denom.
+/// Would also save us some compute units to find authority and other accounts which used to be found by deriving from the seeds.
+#[derive(Debug, Clone)]
+pub struct TransferAccounts<'a> {
+    pub name: TransferAccountNames,
+    pub account: AccountInfo<'a>,
+}
+
+/// The various account names which are used for the transfers.
+#[derive(Debug, Clone, PartialEq)]
+pub enum TransferAccountNames {
+    /// Should be a signer
+    Sender,
+    SenderTokenAccount,
+    ReceiverTokenAccount,
+    TokenMint,
+    EscrowAccount,
+    MintAuthority,
+    TokenProgram,
+}
```

Comment:
I’m not sold on this. Why not make it a struct with those fields? Currently, how do you access TokenMint for example? You need to iterate over the vector?

```rust
pub struct TransferAccounts<'a, 'info> {
    sender: Option<&'a AccountInfo<'info>>,
    sender_oken_account: Option<&'a AccountInfo<'info>>,
    receiver_token_account: Option<&'a AccountInfo<'info>>,
    token_mint: Option<&'a AccountInfo<'info>>,
    escrow_account: Option<&'a AccountInfo<'info>>,
    mint_authority: Option<&'a AccountInfo<'info>>,
    token_program: Option<&'a AccountInfo<'info>>,
}
```

Reply:
Well I'm wondering why I chose this method. Struct is only better. I think my initial thought was to iterate over the keys but if they are named, no point in doing so. I will switch to struct.